### PR TITLE
light cow refactor

### PR DESCRIPTION
**Pull request by JoeWaddell**
### Context

Issue(s) closed by this pull request: no direct issues, just general improvement in typing, documentation, and general adherence to RuFaS principles.

- [x] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.
- [x] The default animal JSON has been reverted to default.

### What

Removed four returns (`estimated_daily_milk_produced`, `fat_percent`, `daily_fat_correct_milk_production`, and `cull_stage`) from `cow.update` that were unused.

Removed three of the above returns from `milking_update`, since they were no longer used in `update`.

Added type hints across files "touched" by this light refactor.  

Changed name of `_cull_cows_and_record_stats` to `_evaluate_and_update_cows` to better reflect what this method does, and to match the naming conventions for other animal classes in `life_cycle`.

### Why

Trying to speed things up, and if not lieu of achieving that goal, make the code cleaner.

### How

Gave careful thought to what returns are necessary, and which returns on main could instead simply be modification of object attributes (most of these returns were already modifying attributes!).

### Test plan

Updated existing unit tests to reflect changes in names and returns.

### CODE REVIEW

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 1]
```diff
@@ -310,7 +310,8 @@ def update_milk_production_history(self, sim_day):
 
         Parameter
         ---------
-            sim_day: simulation day
+        sim_day : int
+            Day of simulation.
         """
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/cow.py]:
```suggestion
            Day of simulation.
            
        """
```

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 2]
```diff
@@ -547,7 +533,7 @@ def calc_manure_excretion(
                 nutrient_conc=nutrient_conc,
             )
 
-    def set_nutrient_rqmts(self, animal_grouping_scenario, nutrient_conc: dict = {}):
+    def set_nutrient_rqmts(self, animal_grouping_scenario, nutrient_conc: dict = {}) -> None:
         """
         Calculates this Cow's nutrient requirements.
         """
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/cow.py]:
```suggestion
        Calculates this Cow's nutrient requirements.
        
        """
```

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 3]
```diff
@@ -588,11 +574,13 @@ def set_nutrient_rqmts(self, animal_grouping_scenario, nutrient_conc: dict = {})
...
+    def phosphorus_rqmts(self, DMI: float) -> None:
         """
         Calculates and sets the animal's phosphorus requirement.
-        Args:
-            DMI: the Dry Matter Intake (kg)
+        Parameters
+        ----------
+        DMI : float
+            The Dry Matter Intake (kg)
         """
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/cow.py]:
```suggestion
            The Dry Matter Intake (kg).
 
        """
```

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 4]
```diff
@@ -635,28 +623,35 @@ def phosphorus_rqmts(self, DMI):
...
 
-    def calc_daily_walking_dist(self, vertical_dist_to_parlor, horizontal_dist_to_parlor):
+    def calc_daily_walking_dist(self, vertical_dist_to_parlor: float, horizontal_dist_to_parlor: float) -> None:
         """
         Calculates and sets the animal's daily vertical and horizontal
         walking distance (DVD and DHD).
-        Args:
-            vertical_dist_to_parlor: vertical distance to milking parlor, km
-            horizontal_dist_to_parlor: horizontal distance to milking parlor, km
+        Parameters
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/cow.py:644]:
```suggestion

        Parameters
```

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 5]
```diff
@@ -635,28 +623,35 @@ def phosphorus_rqmts(self, DMI):
...
+        ----------
+        vertical_dist_to_parlor : flaot
+            Vertical distance to milking parlor, km.
+        horizontal_dist_to_parlor : float
+            Horizontal distance to milking parlor, km.
         """
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/cow.py]:
```suggestion
            Horizontal distance to milking parlor (km).

        """
```

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 6]
```diff
@@ -635,28 +623,35 @@ def phosphorus_rqmts(self, DMI):
...
         Calculates and sets the animal's daily vertical and horizontal
         walking distance (DVD and DHD).
-        Args:
-            vertical_dist_to_parlor: vertical distance to milking parlor, km
-            horizontal_dist_to_parlor: horizontal distance to milking parlor, km
+        Parameters
+        ----------
+        vertical_dist_to_parlor : flaot
+            Vertical distance to milking parlor, km.
+        horizontal_dist_to_parlor : float
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/cow.py]:
```suggestion
            Vertical distance to milking parlor (km).
        horizontal_dist_to_parlor : float
```

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 7]
```diff
@@ -635,28 +623,35 @@ def phosphorus_rqmts(self, DMI):
...
         """
         # multiplied by 2 for return trip
         self.DVD = 2 * vertical_dist_to_parlor * AnimalBase.config["cow_times_milked_per_day"]
         self.DHD = 2 * horizontal_dist_to_parlor * AnimalBase.config["cow_times_milked_per_day"]
 
-    def get_bw_change(self, CI):  # noqa
+    def get_bw_change(self, CI: int | float) -> float:  # noqa
         """
-        life cycle psedocode @[A.1A.C.56/57/58]
+        Life cycle pseudocode @[A.1A.C.56/57/58]
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/cow.py]: Maybe add this to a reference section?

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 8]
```diff
@@ -742,8 +728,10 @@ def update(self, sim_day, calving_interval):  # noqa
             If reproduction program not in list of current reproduction programs.
 
         """
-        if self.culled:
-            return None
+        # if self.culled:
+        #     # TODO: remove this part of the code if never should be used
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/cow.py]: Create/link an issue to this TODO.

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 9]
```diff
@@ -2131,29 +2109,24 @@ def should_decrease_conception_rate_by_parity(self) -> bool:
         return AnimalBase.config["decrease_conception_rate_by_parity"]
 
     # Cull methods
-    def cull_update(self, sim_day):
+    def cull_update(self, sim_day: int) -> None:
         """
         Update culling time and cull reasons for cow to leave the herd
-        The reasons are reproduction failure, low production, and health issues
-        Returns: not culled
+        The reasons are reproduction failure, low production, and health issues.
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/cow.py]:
```suggestion
        The reasons are reproduction failure, low production, and health issues.

```

File: RUFAS/routines/animal/life_cycle/life_cycle.py [hunk 10]
```diff
@@ -779,15 +777,22 @@ def _cull_cows_and_record_stats(
     ) -> int:
         """Culls cows and records stats.
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/life_cycle.py]:
```suggestion
        """
        Culls cows and records stats.

```

File: tests/animal_module_tests/test_animal_life_cycle.py [hunk 11]
```diff
@@ -1316,7 +1316,7 @@ def test_calc_percent_cow_per_parity(mocker: MockerFixture, life_cycle_manager:
             assert life_cycle_manager.percent_cow_for_parity[parity] == approx(0.0)
 
 
-def test_cull_cows_and_record_stats(mocker: MockerFixture, life_cycle_manager: LifeCycleManager) -> None:
+def test_evaluate_and_update_cows(mocker: MockerFixture, life_cycle_manager: LifeCycleManager) -> None:
     # Arrange
```
Comment (matthew7838) [inline tests/animal_module_tests/test_animal_life_cycle.py:1340]: Needs a line of simple docstring explanations for test methods.

**Review by matthew7838 — APPROVED** [review]
LGTM, has some formatting issues. Great PR.

**Review by allisterakun — APPROVED** [review]
Looks good, Joe! Great PR. Thank you!

**Pull request merged**